connector/mqtt/locust/setup.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+# Creates n devices into dojot using a pre-defined template and
+# adds the  device indentifiers into redis
+set -ex
+
+# Dojot parameters
+DOJOT_URL=${DOJOT_URL:-"http://127.0.0.1:8000"}
+DOJOT_USER=${DOJOT_USER:-"admin"}
+DOJOT_PASSWD=${DOJOT_PASSWD:-"admin"}
+
+#Environment
+DOJOT_ENV=${DOJOT_ENV:-"n"}
+
+# Redis parameters
+REDIS_HOST=${REDIS_HOST:-"127.0.0.1"}
+REDIS_PORT=${REDIS_PORT:-"6379"}
+REDIS_PASSWD=${REDIS_PASSWD:-""}
+
+# Devices
+NUMBER_OF_DEVICES=${NUMBER_OF_DEVICES:-"10000"}
+
+if [ "${DOJOT_ENV}" == "y" ]
+then
+  # Get JWT Token
+  echo 'Getting jwt token ...'
+  JWT=$(curl --silent -X POST ${DOJOT_URL}/auth \
+  -H "Content-Type:application/json" \
+  -d "{\"username\": \"${DOJOT_USER}\", \"passwd\" : \"${DOJOT_PASSWD}\"}" | jq '.jwt' | tr -d '"')
+
+  if [ -z "$JWT" ];then
+      echo "--- There's no token! ---"
+      exit 1
+  else
+      echo "... Got jwt token ${JWT}."
+  fi
+
+  # Create Template
+  echo 'Creating template ...'
+  TEMPLATE_ID=$(curl --silent -X POST ${DOJOT_URL}/template \
+  -H 'Content-Type:application/json' \
+  -H "Authorization: Bearer ${JWT}" \
+  -d  '{
+        "label": "CargoContainer",
+        "attrs": [
+                    {
+                      "label": "temperature",
+                      "type": "dynamic",
+                      "value_type": "float"},
+                    {
+                      "label": "humidity",
+                      "type": "dynamic",
+                      "value_type": "float"},
+                    {
+                      "label": "lightness",
+                      "type": "dynamic",
+                      "value_type": "float"},
+                    {
+                      "label" : "gps",
+                      "type" : "dynamic",
+                      "value_type" : "geo:point"}
+                ]
+      }' | jq '.template.id')
+
+  if [ $? -ne 0 ]
+  then
+    echo "Could not create template."
+    exit 1
+  else
+    echo "Create template exit code: $?"
+    echo "... Created template ${TEMPLATE_ID}."
+  fi
+
+    # Get JWT Token
+    echo 'Getting jwt token ...'
+    JWT=$(curl --silent -X POST ${DOJOT_URL}/auth \
+    -H "Content-Type:application/json" \
+    -d "{\"username\": \"${DOJOT_USER}\", \"passwd\" : \"${DOJOT_PASSWD}\"}" | jq '.jwt' | tr -d '"')
+    echo "... Got jwt token ${JWT}."
+
+    # Create Template
+    echo 'Creating template ...'
+    TEMPLATE_ID=$(curl --silent -X POST ${DOJOT_URL}/template \
+    -H 'Content-Type:application/json' \
+    -H "Authorization: Bearer ${JWT}" \
+    -d  "{
+          \"templates\": [ \"${TEMPLATE_ID}\" ],
+          \"attrs\": {},
+          \"label\": \"CargoContainer_${I}\"
+        }" | jq '.devices[].id' | tr -d '"')
+
+    if [ $? -eq 0 ]
```

Comment:
I noticed that the **-e** flag was set in the **set** command at the beginning of the script, this will make the whole script exit if a command fails, so if the last previous command fails `| tr -d '"'`, would the script still arrive at this IF to compare the return of the last command?